utils/golang/utils.go

```diff
@@ -99,17 +100,41 @@ func GetDependenciesGraph(projectDir string) (map[string][]string, error) {
 	return deps, nil
 }
 
-func GetArtifactoryRemoteRepoUrl(serverDetails *config.ServerDetails, repo string) (string, error) {
+type GoProxyUrlParams struct {
+	// Fallback to retrieve the modules directly from the source if
+	// the module failed to be retrieved from the proxy.
+	IsDirect bool
+	// The path between baseUrl to go repo standard path
+	EndpointPrefix string
+}
+
+func (gdu *GoProxyUrlParams) BuildUrl(url *url.URL, repoName string) string {
+	if gdu.EndpointPrefix != "" {
+		url.Path += gdu.EndpointPrefix
+	}
+	url.Path += "api/go/" + repoName
+
+	return url.String()
+}
+
+func (gdu *GoProxyUrlParams) AddDirect(url string) string {
+	if gdu.IsDirect && !strings.HasSuffix(url, "|direct") {
+		return url + "|direct"
+	}
+	return url
+}
+
+func GetArtifactoryRemoteRepoUrl(serverDetails *config.ServerDetails, repo string, goProxyParams GoProxyUrlParams) (string, error) {
 	authServerDetails, err := serverDetails.CreateArtAuthConfig()
 	if err != nil {
 		return "", err
 	}
-	return getArtifactoryApiUrl(repo, authServerDetails)
+	return getArtifactoryApiUrl(repo, authServerDetails, goProxyParams)
 }
 
 // Gets the URL of the specified repository Go API in Artifactory.
 // The URL contains credentials (username and access token or password).
-func getArtifactoryApiUrl(repoName string, details auth.ServiceDetails) (string, error) {
+func getArtifactoryApiUrl(repoName string, details auth.ServiceDetails, goProxyParams GoProxyUrlParams) (string, error) {
```

Comment:
You enhanced this function by adding a new `goProxyParams` argument to it, however, you did not enhance the existing `TestGetArtifactoryApiUrl` function to also test new argument functionality.